provider/provider.go

```diff
@@ -104,13 +105,16 @@ type Tile interface {
 	BufferedExtent() (extent *geom.Extent, srid uint64)
 }
 
+// ParameterTokenRegexp to validate QueryParameters
+var ParameterTokenRegexp = regexp.MustCompile("![a-zA-Z0-9_-]+!")
```

Comment:
If we go with my suggestion of using the param name as the token name, Then I'd allow `.`, too:
```suggestion
var ParameterTokenRegexp = regexp.MustCompile("![a-zA-Z0-9._-]+!")
```